library/core/src/hash/mod.rs

```diff
@@ -234,12 +235,19 @@ pub trait Hash {
     /// [`hash`]: Hash::hash
     /// [`hash_slice`]: Hash::hash_slice
     #[stable(feature = "hash_slice", since = "1.3.0")]
-    fn hash_slice<H: Hasher>(data: &[Self], state: &mut H)
+    fn hash_slice<H: ~const Hasher>(data: &[Self], state: &mut H)
     where
         Self: Sized,
     {
-        for piece in data {
-            piece.hash(state);
+        //FIXME(const_iter_slice): Revert to for loop
+        //for piece in data {
+        //    piece.hash(state);
+        //}
+
+        let mut i = 0;
+        while i < data.len() {
+            data[i].hash(state);
+            i += 1;
         }
```

Comment:
`data[i]` has bound checks IIRC. you could try writing a benchmark and see if its performance has changed or use `const_eval_select` to reduce runtime performance impact.

Reply:
> Left some nits. Also, I would like to see some tests to ensure that it works (maybe not as an ui test, but under `library/core/tests`, next to the hash tests)

Should I add a new file for testing const Hash or should i make the current Tests const and add some asserts that compare the CT and RT results?

Reply:
I used const_eval_select for now.
I hope that const `Iterator` for `Slice` won't take to long.